ConcurrentModificationException when querying ClickHouse table (flaky test)

https://github.com/trinodb/trino/actions/runs/4681805913/jobs/8298469961?pr=16995

```
Error:  Tests run: 707, Failures: 1, Errors: 0, Skipped: 158, Time elapsed: 399.179 s <<< FAILURE! - in TestSuite
Error:  io.trino.plugin.clickhouse.TestClickHouseLatestTypeMapping.testClickHouseDateTimeMinMaxValues[1970-01-01 00:00:00](1)  Time elapsed: 0.023 s  <<< FAILURE!
java.lang.RuntimeException: 
Error executing sql:
CREATE TABLE test_timestampmte42r6t38 AS SELECT CAST(timestamp '1970-01-01 00:00:00' AS timestamp(0)) col_0
	at io.trino.testing.sql.TrinoSqlExecutor.execute(TrinoSqlExecutor.java:45)
	at io.trino.testing.sql.TestTable.createAndInsert(TestTable.java:56)
	at io.trino.testing.sql.TestTable.<init>(TestTable.java:44)
	at io.trino.testing.sql.TestTable.<init>(TestTable.java:36)
	at io.trino.testing.datatype.CreateAsSelectDataSetup.setupTemporaryRelation(CreateAsSelectDataSetup.java:46)
	at io.trino.testing.datatype.CreateAsSelectDataSetup.setupTemporaryRelation(CreateAsSelectDataSetup.java:25)
	at io.trino.testing.datatype.SqlDataTypeTest.execute(SqlDataTypeTest.java:90)
	at io.trino.plugin.clickhouse.BaseClickHouseTypeMapping.testClickHouseDateTimeMinMaxValues(BaseClickHouseTypeMapping.java:738)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.trino.testing.QueryFailedException: Unknown error 1002, server ClickHouseNode [uri=http://localhost:32772/default, options={use_binary_string=true}]@-768217720
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
	at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:495)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:488)
	at io.trino.testing.sql.TrinoSqlExecutor.execute(TrinoSqlExecutor.java:42)
	... 20 more
	Suppressed: java.lang.Exception: SQL: CREATE TABLE test_timestampmte42r6t38 AS SELECT CAST(timestamp '1970-01-01 00:00:00' AS timestamp(0)) col_0
		at io.trino.testing.DistributedQueryRunner.executeWithQueryId(DistributedQueryRunner.java:499)
		... 22 more
Caused by: io.trino.spi.TrinoException: Unknown error 1002, server ClickHouseNode [uri=http://localhost:32772/default, options={use_binary_string=true}]@-768217720
	at io.trino.plugin.jdbc.JdbcPageSink.<init>(JdbcPageSink.java:124)
	at io.trino.plugin.jdbc.JdbcPageSinkProvider.createPageSink(JdbcPageSinkProvider.java:45)
	at io.trino.split.PageSinkManager.createPageSink(PageSinkManager.java:49)
	at io.trino.operator.TableWriterOperator$TableWriterOperatorFactory.createPageSink(TableWriterOperator.java:122)
	at io.trino.operator.TableWriterOperator$TableWriterOperatorFactory.createOperator(TableWriterOperator.java:116)
	at io.trino.operator.DriverFactory.createDriver(DriverFactory.java:106)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunnerFactory.createDriver(SqlTaskExecution.java:642)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:834)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:165)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:537)
	at io.trino.$gen.Trino_testversion____20230412_235227_490.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.sql.SQLException: Unknown error 1002, server ClickHouseNode [uri=http://localhost:32772/default, options={use_binary_string=true}]@-768217720
	at com.clickhouse.jdbc.SqlExceptionUtils.handle(SqlExceptionUtils.java:85)
	at com.clickhouse.jdbc.SqlExceptionUtils.create(SqlExceptionUtils.java:31)
	at com.clickhouse.jdbc.SqlExceptionUtils.handle(SqlExceptionUtils.java:90)
	at com.clickhouse.jdbc.internal.ClickHouseConnectionImpl.getTableColumns(ClickHouseConnectionImpl.java:259)
	at com.clickhouse.jdbc.internal.ClickHouseConnectionImpl.prepareStatement(ClickHouseConnectionImpl.java:767)
	at com.clickhouse.jdbc.ClickHouseConnection.prepareStatement(ClickHouseConnection.java:121)
	at io.trino.plugin.jdbc.ForwardingConnection.prepareStatement(ForwardingConnection.java:54)
	at io.trino.plugin.jdbc.ForwardingConnection.prepareStatement(ForwardingConnection.java:54)
	at io.trino.plugin.jdbc.ForwardingConnection.prepareStatement(ForwardingConnection.java:54)
	at io.trino.plugin.jdbc.JdbcPageSink.<init>(JdbcPageSink.java:117)
	... 13 more
Caused by: java.util.ConcurrentModificationException
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1221)
	at com.clickhouse.data.format.BinaryDataProcessor$DateTime32SerDe.of(BinaryDataProcessor.java:219)
	at com.clickhouse.data.format.ClickHouseRowBinaryProcessor.getDeserializer(ClickHouseRowBinaryProcessor.java:361)
	at com.clickhouse.data.ClickHouseDataProcessor.<init>(ClickHouseDataProcessor.java:361)
	at com.clickhouse.data.format.ClickHouseRowBinaryProcessor.<init>(ClickHouseRowBinaryProcessor.java:320)
	at com.clickhouse.data.ClickHouseDataStreamFactory.getProcessor(ClickHouseDataStreamFactory.java:140)
	at com.clickhouse.client.ClickHouseStreamResponse.<init>(ClickHouseStreamResponse.java:74)
	at com.clickhouse.client.ClickHouseStreamResponse.of(ClickHouseStreamResponse.java:51)
	at com.clickhouse.client.http.ClickHouseHttpClient.send(ClickHouseHttpClient.java:128)
	at com.clickhouse.client.AbstractClient.execute(AbstractClient.java:298)
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.sendOnce(ClickHouseClientBuilder.java:282)
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.send(ClickHouseClientBuilder.java:294)
	at com.clickhouse.client.ClickHouseClientBuilder$Agent.execute(ClickHouseClientBuilder.java:349)
	at com.clickhouse.client.ClickHouseClient.executeAndWait(ClickHouseClient.java:1025)
	at com.clickhouse.client.ClickHouseRequest.executeAndWait(ClickHouseRequest.java:2074)
	at com.clickhouse.jdbc.internal.ClickHouseConnectionImpl.getTableColumns(ClickHouseConnectionImpl.java:256)
	... 19 more

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    TestClickHouseLatestTypeMapping>BaseClickHouseTypeMapping.testClickHouseDateTimeMinMaxValues:738 » Runtime
[INFO] 
Error:  Tests run: 707, Failures: 1, Errors: 0, Skipped: 158
[INFO] 
```

--- a/docs/src/main/sphinx/connector/clickhouse.rst
+++ b/docs/src/main/sphinx/connector/clickhouse.rst
@@ -40,7 +40,7 @@ connection properties as appropriate for your setup:
 The ``connection-url`` defines the connection information and parameters to pass
 to the ClickHouse JDBC driver. The supported parameters for the URL are
 available in the `ClickHouse JDBC driver configuration
-<https://github.com/ClickHouse/clickhouse-jdbc/tree/master/clickhouse-jdbc#Configuration>`_.
+<https://clickhouse.com/docs/en/integrations/java#configuration>`_.
 
 The ``connection-user`` and ``connection-password`` are typically required and
 determine the user credentials for the connection, often a service user. You can

--- a/pom.xml
+++ b/pom.xml
@@ -1056,14 +1056,8 @@
             <dependency>
                 <groupId>com.clickhouse</groupId>
                 <artifactId>clickhouse-jdbc</artifactId>
-                <version>0.4.1</version>
+                <version>0.4.6</version>
                 <classifier>all</classifier>
-                <exclusions>
-                    <exclusion>
-                        <groupId>*</groupId>
-                        <artifactId>*</artifactId>
-                    </exclusion>
-                </exclusions>
             </dependency>
 
             <dependency>

